edge/pkg/metamanager/metaserver/handlerfactory/handler.go

```diff
@@ -77,10 +88,14 @@ func (f *Factory) Create(req *request.RequestInfo) http.Handler {
 }
 
 func (f *Factory) Delete() http.Handler {
+	f.lock.RLock()
+	defer f.lock.RUnlock()
 	if h, ok := f.handlers["delete"]; ok {
 		return h
 	}
 	h := handlers.DeleteResource(f.storage, false, f.scope, fakers.NewAlwaysAdmit())
+	f.lock.Lock()
```

Comment:
deadlock

Reply:
@gccio 
ok, modified the lock type, and only locked once inside the function.
/ptal

Reply:
RWMutex lock is still used, and the lock is manually released to prevent deadlock caused by re-entry lock